src/app/qgsidentifyresultsdialog.cpp

```diff
@@ -737,6 +737,9 @@ QgsIdentifyResultsFeatureItem *QgsIdentifyResultsDialog::createFeatureItem( QgsV
     const QString representedValue = representValue( vlayer, setup, fields.at( i ).name(), attrs.at( i ) );
     attrItem->setSortData( 1, representedValue );
     attrItem->setToolTip( 1, representedValue );
+    // To make sure we can copy to clipboard even when an alternate widget is used (json/url),
+    // we store the representedValue to UserRole + 2 and read from that when copying to clipboard
+    attrItem->setData( 1, Qt::UserRole + 2, representedValue );
```

Comment:
it would be test to have the following declaration at top of file and user it wherever Qt::UserRole + 2 is used
```
constexpr int ROLE_REPRESENTED_VALUE = Qt::UserRole + 2;
```

Reply:
Done, this is indeed cleaner!